### PR TITLE
Implement KHR_materials_unlit

### PR DESCRIPTION
**Pull request by donmccurdy**
Based on https://github.com/KhronosGroup/glTF/pull/1163.

Example: [unlit_monkey.gltf.zip](https://github.com/KhronosGroup/glTF-Blender-Exporter/files/1735864/unlit_monkey.gltf.zip)

Demo viewer with unlit support: https://khr-materials-unlit-ooddguuati.now.sh/

### CODE REVIEW

**Comment by donmccurdy** [comment]
On second thought maybe there should not be an `Export KHR_materials_unlit` option in the UI, we can just infer automatically if the "Shadeless" checkbox on the material is enabled in Blender?

**Comment by fernandojsg** [comment]
@donmccurdy +1 for using the `shadedless` option

**Comment by donmccurdy** [comment]
Ok thanks, updated. 👍

File: scripts/addons/io_scene_gltf2/gltf2_generate.py [hunk 1]
```diff
@@ -2561,87 +2564,29 @@ def generate_materials(operator,
                     else:
                         alphaMode = 'BLEND'
 
-                common['diffuseFactor'] = [blender_material.diffuse_color[0] * blender_material.diffuse_intensity,
+                pbrMetallicRoughness['baseColorFactor'] = [blender_material.diffuse_color[0] * blender_material.diffuse_intensity,
```
Comment (fernandojsg) [inline scripts/addons/io_scene_gltf2/gltf2_generate.py]: what about adding default fallback pbt parameters? `metallicFactor: 0, roughnessFactor: 0.5`

Comment (donmccurdy) [reply scripts/addons/io_scene_gltf2/gltf2_generate.py]: Good point; I think `roughnessFactor >= 0.5` is the key thing, so subjectively something like ~0.9 would be about right..

Comment (fernandojsg) [reply scripts/addons/io_scene_gltf2/gltf2_generate.py]: :+1:

Comment (fernandojsg) [reply scripts/addons/io_scene_gltf2/gltf2_generate.py]: In any case it should be consistent with the material_unlit extension where the recommended value for roughness is 0.5, so you could probabliy specify there >=0.5 or set directly 0.9

Comment (donmccurdy) [reply scripts/addons/io_scene_gltf2/gltf2_generate.py]: I don't think we're recommending 0.5 anywhere in the extension... it says just:

> * `roughnessFactor` is greater than `0.5`.

Comment (fernandojsg) [reply scripts/addons/io_scene_gltf2/gltf2_generate.py]: Ah you're right I just misread that part :)

**Comment by fernandojsg** [comment]
Cooool! 👍

**Pull request merged**